corehq/apps/domain/decorators.py

```diff
@@ -416,6 +425,20 @@ def api_auth(*, allow_creds_in_data=True, oauth_scopes=None):
         default=DIGEST,
         oauth_scopes=oauth_scopes,
         allow_creds_in_data=allow_creds_in_data,
+        allow_api_key_as_password=False,  # if supporting digest auth, you cannot use an api key as a password
+    )
+
+
+def api_auth_no_digest(*, allow_creds_in_data=True, oauth_scopes=None, allow_api_key_as_password=False):
```

Comment:
Is the `allow_api_key_as_password=False` parameter needed here? The only place it is ever `False` is in tests. What do you think of eliminating it and always passing `allow_api_key_as_password=True` to `get_multi_auth_decorator` below?

Since this function is about allowing API key as password, would it make sense to rename it to make that more clear and instead explain the part about "no digest" in the docstring? Maybe something like `api_auth_allow_key_as_password`?

Reply:
986fd8637112bd1b9e1fd95f247429080c9e70c1